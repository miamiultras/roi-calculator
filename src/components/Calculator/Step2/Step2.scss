.step2-container {
  width: 100%;
  color: #fff;
  h1 {
    font-family: 'Roboto';
    font-weight: 300;
    color: #fff;
    font-size: 36px;
    margin: 20px 0 10px 0;
    text-transform: uppercase;
  }

  img {
    margin-top: 34px;
    height: 76px;
  }

  input[type="text"] {
    width: 242px;
    padding: 4px 8px;
    text-align: center;
    color: #999;
    font-size: 21px;
    margin-bottom: 50px;
  }
}
