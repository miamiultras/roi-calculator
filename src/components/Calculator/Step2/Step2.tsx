import React from 'react';
import './Step2.scss';

import icon from '../../../assets/images/diagram.svg';
import Paragraph from '../../Paragraph/Paragraph';
import NextButton from '../../NextButton/NextButton';
import BackButton from '../../BackButton/BackButton';
import CalculatorNavigation from '../../CalculatorNavigation/CalculatorNavigation';

const Step2 = ({ estimatedSellInput, handleChange }: { estimatedSellInput: string, handleChange: any }) => (
  <div className='step2-container'>
    <div className='title'>
      <h1>Live PPV Event ROI Calculator</h1>
    </div>
    <Paragraph>
      Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
    </Paragraph>
    <img src={icon} alt='icon' />
    <Paragraph>
      How many tickets do you estimate to sell?
    </Paragraph>

    <input
      id='estimatedSellInput'
      type='text'
      name='estimatedSellInput'
      value={estimatedSellInput}
      onChange={handleChange}
    />

    <NextButton name='next question' to='step3' />
    <BackButton name='back' to='step1' />
    <CalculatorNavigation step={2} />
  </div>
);

export default Step2;
