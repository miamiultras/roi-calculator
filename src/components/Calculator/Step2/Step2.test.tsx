import * as React from 'react';
import { shallow } from 'enzyme';
import Step2 from './Step2';

function functionMock() { }

function setup() {
  const props = {};
  const wrapper = shallow(
    <Step2 estimatedSellInput='15000' handleChange={functionMock} />
  );
  return {
    wrapper,
    props,
  };
}

describe('Step2 component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
