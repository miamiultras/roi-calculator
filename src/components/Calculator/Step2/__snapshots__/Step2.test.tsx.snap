// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step2 component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Step2
    estimatedSellInput="15000"
    handleChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          <h1>
            Live PPV Event ROI Calculator
          </h1>
        </div>,
        <Paragraph>
          Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
        </Paragraph>,
        <img
          alt="icon"
          src="diagram.svg"
        />,
        <Paragraph>
          How many tickets do you estimate to sell?
        </Paragraph>,
        <input
          id="estimatedSellInput"
          name="estimatedSellInput"
          onChange={[Function]}
          type="number"
          value="15000"
        />,
        <NextButton
          name="next question"
          to="step3"
        />,
        <BackButton
          name="back"
          to="step1"
        />,
        <CalculatorNavigation
          step={2}
        />,
      ],
      "className": "step2-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Live PPV Event ROI Calculator
          </h1>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live PPV Event ROI Calculator",
          },
          "ref": null,
          "rendered": "Live PPV Event ROI Calculator",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
        },
        "ref": null,
        "rendered": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "icon",
          "src": "diagram.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "How many tickets do you estimate to sell?",
        },
        "ref": null,
        "rendered": "How many tickets do you estimate to sell?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "estimatedSellInput",
          "name": "estimatedSellInput",
          "onChange": [Function],
          "type": "number",
          "value": "15000",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "next question",
          "to": "step3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "back",
          "to": "step1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "step": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            <h1>
              Live PPV Event ROI Calculator
            </h1>
          </div>,
          <Paragraph>
            Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
          </Paragraph>,
          <img
            alt="icon"
            src="diagram.svg"
          />,
          <Paragraph>
            How many tickets do you estimate to sell?
          </Paragraph>,
          <input
            id="estimatedSellInput"
            name="estimatedSellInput"
            onChange={[Function]}
            type="number"
            value="15000"
          />,
          <NextButton
            name="next question"
            to="step3"
          />,
          <BackButton
            name="back"
            to="step1"
          />,
          <CalculatorNavigation
            step={2}
          />,
        ],
        "className": "step2-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Live PPV Event ROI Calculator
            </h1>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Live PPV Event ROI Calculator",
            },
            "ref": null,
            "rendered": "Live PPV Event ROI Calculator",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
          },
          "ref": null,
          "rendered": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "icon",
            "src": "diagram.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "How many tickets do you estimate to sell?",
          },
          "ref": null,
          "rendered": "How many tickets do you estimate to sell?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "estimatedSellInput",
            "name": "estimatedSellInput",
            "onChange": [Function],
            "type": "number",
            "value": "15000",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "next question",
            "to": "step3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "back",
            "to": "step1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "step": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
